graylog2-server/src/main/java/org/graylog2/messageprocessors/MessageProcessorOrder.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.messageprocessors;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.AutoValue;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+@AutoValue
+@JsonAutoDetect
+public abstract class MessageProcessorOrder {
+
+    @JsonProperty
+    public abstract long changeVersion();
+
+    @JsonProperty
+    public abstract List<String> classOrder();
+
+    @JsonProperty
+    public abstract Set<String> disabledMessageProcessors();
+
+    @JsonCreator
+    public static MessageProcessorOrder create(@JsonProperty("changeVersion") long changeVersion,
+                                               @JsonProperty("classOrder") List<String> classOrder,
+                                               @JsonProperty("disabledMessageProcessors") Set<String> disabledMessageProcessors) {
+        return new AutoValue_MessageProcessorOrder(changeVersion, classOrder, disabledMessageProcessors);
+    }
+
+    @JsonCreator
```

Comment:
I'm not sure about the behavior of Jackson with multiple `@JsonCreator` annotations in a class. I think it'll only ever try to use the first one.